### PR TITLE
REF: use extract_array in DataFrame.combine_first

### PR DESCRIPTION
**Pull request by arw2019**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
lgtm cc @jbrockmendel

**Comment by jbrockmendel** [comment]
the extract_array part looks good, but not clear why the needs_i8_conversion part can be removed safely

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6369,29 +6368,11 @@ def combine_first(self, other: DataFrame) -> DataFrame:
...
-            mask = isna(x)
-            # TODO: extract_array?
-            if isinstance(mask, (Index, Series)):
-                mask = mask._values
+            mask = extract_array(isna(x))
 
-            x_values = extract_values(x)
-            y_values = extract_values(y)
+            x_values = extract_array(x)
+            y_values = extract_array(y)
```
Comment (jbrockmendel) [inline pandas/core/frame.py:6375]: i guess pass `extract_numpy=True` on these

Comment (arw2019) [reply pandas/core/frame.py:6375]: Done

**Comment by arw2019** [comment]
> the extract_array part looks good, but not clear why the needs_i8_conversion part can be removed safely

I reverted it

Although CI is not passing without for lack of testing with datelike input - AFAICT we have good (but maybe not water-tight) coverage there

**Comment by jbrockmendel** [comment]
> Although CI is not passing without for lack of testing with datelike input - AFAICT we have good (but maybe not water-tight) coverage there

it may be unnecessary (in fact i hope it is), just want to have an idea why its unnecessary (or at least make sure we have relevant test coverage)

**Comment by arw2019** [comment]
> > Although CI is not passing without for lack of testing with datelike input - AFAICT we have good (but maybe not water-tight) coverage there
> 
> it may be unnecessary (in fact i hope it is), just want to have an idea why its unnecessary (or at least make sure we have relevant test coverage)

Ok! Checked that both parts of the if/else clause inside `if needs_i8_conversion(arr.dtype)` are hit in:
```
pandas/tests/frame/methods/test_combine_first.py::TestDataFrameCombineFirst::test_combine_first_timezone
pandas/tests/frame/methods/test_combine_first.py::TestDataFrameCombineFirst::test_combine_first_period
```

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**